app/src/ui/changes/commit-message.tsx

```diff
@@ -423,6 +424,13 @@ export class CommitMessage extends React.Component<
     }
   }
 
+  private isGitConfigLocal = async () => {
+    const { repository } = this.props
+    const localName = await getConfigValue(repository, 'user.name', true)
+    const localEmail = await getConfigValue(repository, 'user.email', true)
+    return localName !== null || localEmail !== null
+  }
+
```

Comment:
In the commit avatar component, I already had logic to run this in the constructor and then on `componentDidUpdate` (didn't know that having in the props would cause it to run),  if the user name or email had changed as I figured that was a decent indication of when git configuration changed.. tho not fool proof as someone could have the same name and email in global and in local. 

I don't think on app focus would be frequent enough because I need the change to pick up if the user goes to the repository settings and modifies them so that the avatar shows the correct message.

Reply:
I don't think it is that important for this to be fool proof as it is just whether the user is directed to the repository settings or git user settings.. and to be honest, I doubt this will be a high traffic area to the extent it is used it will likely be accurate. But, if anyone feels it should be better.. we could track when the git config location changes on the repository state to send down to the avatar component.